### DISCUSSION TITLE
Connect AWS Object Store to MinIO

### DISCUSSION THREAD

**Discussion by gianarb** [apache/arrow-rs#4462] [Q&A] [answered]
Hello

I would like to use MinIO as endpoint for the AWS Object Store and in order to do that I am using the `AWSBuilder::from_env`, but there is an http request that tries to authenticate using my token that tries to reach the aws metadata api, and since I am not on AWS it fails. I can't tell what I am doing wrong or if this is not possible. Thanks

```rust
    let s3 = match AmazonS3Builder::from_env()
        .with_bucket_name(&bucket_name)
        .with_region(region)
        .build()
```

```
AWS_ENDPOINT=http://127.0.0.1:9000
AWS_ACCESS_KEY_ID=admin
AWS_SECRET_ACCESS_KEY=admin
AWS_ALLOW_HTTP=true
```

```
        message: "request error",
        source: Some(
            reqwest::Error {
                kind: Request,
                url: Url {
                    scheme: "http",
                    cannot_be_a_base: false,
                    username: "",
                    password: None,
                    host: Some(
                        Ipv4(
                            169.254.169.254,
                        ),
                    ),
                    port: None,
                    path: "/latest/api/token",
                    query: None,
                    fragment: None,
                },
                source: hyper::Error(
                    Connect,
                    ConnectError(
                        "tcp connect error",
                        Os {
                            code: 110,
                            kind: TimedOut,
                            message: "Connection timed out",
                        },
                    ),
                ),
            },
        ),
    },
}
```

**Comment by tustvold** [comment, marked as answer]
The error would suggest it is failing to find credentials and falling back to trying to get them from a metadata endpoint. Are you sure the environment variables are reaching the process? Perhaps you could try setting the credentials explicitly e.g. with https://docs.rs/object_store/latest/object_store/aws/struct.AmazonS3Builder.html#method.with_access_key_id

**Comment by gianarb** [comment]
You are right, I was not exposing the environment variables as I thought I was doing :crab: sorry for wasting your time and thank you for your help

**Reply by tustvold** [reply]
No worries, glad you got it working
